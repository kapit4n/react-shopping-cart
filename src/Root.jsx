import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import AppLegacy from "./app-legacy/AppLegacy";
import App from "./app/App";

const displayNewApp = true

const Root = ({ store }) => (
  <Provider store={store}>
    <Router>
      {displayNewApp ? <App />: <AppLegacy />}
    </Router>
  </Provider>
);

Root.propTypes = {
  store: PropTypes.object.isRequired
};

export default Root;
