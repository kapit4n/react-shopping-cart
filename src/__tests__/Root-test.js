import React from 'react';
import Root from '../Root'

import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from "react-router-dom";

import Enzyme, { shallow, mount } from 'enzyme';
// Import Adapter
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
//use adapter
Enzyme.configure({ adapter: new Adapter() });

const store = createStore(r => r, {
  products: {
    todoApp: {
      products: [],
      cartInfo: {
        total: 10
      },
      filters: {
        display: true
      }
    }
  }
})

describe('shallow render Root', () => {
  const wrapper = shallow(<Root store={store}/>)
  it('', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mount render Root', () => {
  const wrapper = mount(
        <Root store={store}/>
  )
  it('', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

