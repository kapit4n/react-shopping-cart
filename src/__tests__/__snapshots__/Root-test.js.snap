// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount render App  1`] = `ReactWrapper {}`;

exports[`mount render Root  1`] = `ReactWrapper {}`;

exports[`shallow render App  1`] = `ShallowWrapper {}`;

exports[`shallow render Root  1`] = `ShallowWrapper {}`;
